EventFilter/EcalRawToDigi/plugins/EcalCPUDigisProducer.cc

```diff
@@ -53,17 +70,55 @@ void EcalCPUDigisProducer::fillDescriptions(edm::ConfigurationDescriptions& conf
   desc.add<std::string>("digisOutLabelEB", "ebDigis");
   desc.add<std::string>("digisOutLabelEE", "eeDigis");
 
+  desc.add<bool>("produceDummyIntegrityCollections", false);
+
   std::string label = "ecalCPUDigisProducer";
   confDesc.add(label, desc);
 }
 
 EcalCPUDigisProducer::EcalCPUDigisProducer(const edm::ParameterSet& ps)
-    : digisInEBToken_{consumes<cms::cuda::Product<ecal::DigisCollection>>(
-          ps.getParameter<edm::InputTag>("digisInLabelEB"))},
+    :  // input digi collections in GPU-friendly format
+      digisInEBToken_{
+          consumes<cms::cuda::Product<ecal::DigisCollection>>(ps.getParameter<edm::InputTag>("digisInLabelEB"))},
       digisInEEToken_{
           consumes<cms::cuda::Product<ecal::DigisCollection>>(ps.getParameter<edm::InputTag>("digisInLabelEE"))},
+      // output digi collections in legacy format
       digisOutEBToken_{produces<EBDigiCollection>(ps.getParameter<std::string>("digisOutLabelEB"))},
-      digisOutEEToken_{produces<EEDigiCollection>(ps.getParameter<std::string>("digisOutLabelEE"))} {}
+      digisOutEEToken_{produces<EEDigiCollection>(ps.getParameter<std::string>("digisOutLabelEE"))},
+      // whether to produce dummy integrity collections
+      produceDummyIntegrityCollections_{ps.getParameter<bool>("produceDummyIntegrityCollections")},
+      // dummy SRP collections
+      ebSrFlagToken_{produceDummyIntegrityCollections_ ? produces<EBSrFlagCollection>()
+                                                       : edm::EDPutTokenT<EBSrFlagCollection>{}},
+      eeSrFlagToken_{produceDummyIntegrityCollections_ ? produces<EESrFlagCollection>()
+                                                       : edm::EDPutTokenT<EESrFlagCollection>{}},
```

Comment:
The same could be achieved in the constructor body with a bit less typing with
```cpp
if (produceDummyIntegrityCollections_) {
  ebSrFlagToken_ = produces<EBSrFlagCollection>();
  eeSrFlagToken_ = produces<EESrFlagCollection>();
  ...
}
```

Reply:
I know, but I usually prefer to put the initialisations in the initialiser list (though I expect for these cases with decent optimisations there is no difference in the generated code)